rust/src/bin/test/MyGame/Example/Monster.rs

```diff
@@ -0,0 +1,534 @@
+//! Automatically generated, do not modify.
+
+use flatbuffers;
+use super::*;
+
+/// an example documentation comment: monster object
+#[derive(Debug)]
+pub struct Monster<'a> {
+    table: flatbuffers::Table<'a>,
+}
+
+enum VT {
+    POS = 4,
+    MANA = 6,
+    HP = 8,
+    NAME = 10,
+    INVENTORY = 14,
+    COLOR = 16,
+    TEST_TYPE = 18,
+    TEST = 20,
+    TEST4 = 22,
+    TESTARRAYOFSTRING = 24,
+    TESTARRAYOFTABLES = 26,
+    ENEMY = 28,
+    TESTNESTEDFLATBUFFER = 30,
+    TESTEMPTY = 32,
+    TESTBOOL = 34,
+    TESTHASHS32_FNV1 = 36,
+    TESTHASHU32_FNV1 = 38,
+    TESTHASHS64_FNV1 = 40,
+    TESTHASHU64_FNV1 = 42,
+    TESTHASHS32_FNV1A = 44,
+    TESTHASHU32_FNV1A = 46,
+    TESTHASHS64_FNV1A = 48,
+    TESTHASHU64_FNV1A = 50,
+    TESTARRAYOFBOOLS = 52,
+    TESTF = 54,
+    TESTF2 = 56,
+    TESTF3 = 58,
+}
+
+impl<'a> Monster<'a> {
+    pub fn new(buf: &[u8], offset: flatbuffers::UOffsetT) -> Monster {
+        Monster {
+            table: flatbuffers::Table::from_offset(buf, offset),
+        }
+    }
+
+    pub fn pos(&self) -> Option<Vec3> {
+        let offset = self.table.field_offset(VT::POS as u16);
```

Comment:
can this code pattern be abstracted? no need for the generated code to be any more repetitive than need be